### PR TITLE
fix(project-access): app folder bug, no new projects into CAP folder

### PR DESCRIPTION
**Pull request by korotkovao**
Fix the bug for the `app` folder within CAP as the target folder for the new project.

Alert and prevent from generating a Fiori application into a CAP project folder unless I have chosen the Local CAP Project as my data source.

### CODE REVIEW

**Review by devinea — APPROVED** [review]
Code changes look ok. 
@Klaus-Keller Please review also.

File: packages/project-access/src/project/search.ts [hunk 1]
```diff
@@ -228,7 +228,7 @@ export async function findCapProjectRoot(path: string): Promise<string | null> {
             if (await getCapProjectType(projectRoot)) {
                 // We have found a CAP project as root. Check if the found app is not directly in CAP's 'app/' folder.
                 // Sometime there is a <CAP_ROOT>/app/package.json file that is used for app router (not an app)
-                if (join(projectRoot, 'app') !== path) {
+                if (join(projectRoot, 'app') !== path || join(projectRoot, 'app/') !== path) {
```
Comment (longieirl) [inline packages/project-access/src/project/search.ts:231]: Have you tested this on windows? My suggestion would be;
```
if (join(projectRoot, 'app') !== path || join(projectRoot, 'app', path.sep) !== path) {
```

File: packages/project-access/src/project/search.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { basename, dirname, isAbsolute, join, parse, sep } from 'path';
+import path from 'path';
```
Comment (Klaus-Keller) [inline packages/project-access/src/project/search.ts:2]: Why do you import `path` here? this leads to a [codesmell](https://sonarcloud.io/project/issues?id=SAP_open-ux-tools&pullRequest=2438&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) and `sep` is already imported. Then you could also avoid renaming the path varible.

File: packages/project-access/src/project/search.ts [hunk 3]
```diff
@@ -214,21 +215,21 @@ export async function findRootsForPath(path: string): Promise<{ appRoot: string;
...
-        const { root } = parse(path);
-        let projectRoot = dirname(path);
+        const { root } = parse(dir);
+        let projectRoot = dirname(dir);
         while (projectRoot !== root) {
             if (await getCapProjectType(projectRoot)) {
                 // We have found a CAP project as root. Check if the found app is not directly in CAP's 'app/' folder.
                 // Sometime there is a <CAP_ROOT>/app/package.json file that is used for app router (not an app)
-                if (join(projectRoot, 'app') !== path) {
+                if (join(projectRoot, 'app') !== dir || join(projectRoot, 'app', path.sep) !== dir) {
```
Comment (Klaus-Keller) [inline packages/project-access/src/project/search.ts:232]:
```suggestion
                if (join(projectRoot, 'app') !== dir || join(projectRoot, 'app', sep) !== dir) {
```

File: packages/project-access/src/project/search.ts (hunk 2 above)
Comment (korotkovao) [reply packages/project-access/src/project/search.ts]: Thank you, I have updated the code.

File: packages/project-access/src/project/search.ts (hunk 3 above)
Comment (korotkovao) [reply packages/project-access/src/project/search.ts]: Thank you, I have updated the code.

File: packages/project-access/test/project/cap.test.ts [hunk 4]
```diff
@@ -104,6 +104,13 @@ describe('Test isPathForCapApp()', () => {
         expect(result).toBe(true);
     });
 
+    test('Should return true for sub directory CAP path, specifially app folder', async () => {
+        const result = await isPathForCapApp(
+            join(__dirname, '../test-data/project/find-all-apps/CAP/CAPnode_fiori_elements/app')
```
[the remark this answers is not in the document]
Comment (korotkovao) [reply packages/project-access/test/project/cap.test.ts:109]: Thank you for the review.
I know it looks counter-intuitive, however, the tailing separator was already passing before the bug fix.
I can add it too.

File: packages/project-access/test/project/cap.test.ts [hunk 5]
```diff
@@ -104,6 +104,20 @@ describe('Test isPathForCapApp()', () => {
...
+        );
+        expect(result).toBe(true);
+    });
+
+    test('Should return true for sub directory CAP path, specifially app/ folder', async () => {
```
Comment (longieirl) [inline packages/project-access/test/project/cap.test.ts:114]: Does it make sense to add tests to handle a windows path?

**Review by Klaus-Keller — APPROVED** [review]
Thanks @korotkovao! 
- fix to find CAP project root looks good.
- changeset is valid
- test coverage is great
- did no manual test, just visual review

Approved from my side

**Review by devinea — APPROVED** [review]

**Review by IainSAP — APPROVED** [review]
Thank you @korotkovao 

- Re-review, comments addressed

**Comment by korotkovao** [comment]
Thank you @devinea @Klaus-Keller @longieirl @IainSAP

**Pull request merged**